semgrep-core/parsing/Parse_go_tree_sitter.ml

```diff
@@ -0,0 +1,1547 @@
+ * file license.txt for more details.
+ *)
+open Common
+module AST = Ast_go
+module CST = Tree_sitter_go.CST
+module PI = Parse_info
+(* open Ast_go *)
+module G = AST_generic
+module H = Parse_tree_sitter_helpers
+
+(*****************************************************************************)
+(* Prelude *)
+(*****************************************************************************)
+(* Go parser using ocaml-tree-sitter-lang/go and converting
+ * to pfff/lang_go/parsing/ast_go.ml
+ *
+ * The resulting AST can then be converted to the generic AST by using
+ * pfff/lang_go/analyze/go_to_generic.ml
+ *)
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+(*****************************************************************************)
+(* Boilerplate converter *)
+(*****************************************************************************)
+(* This was started by copying ocaml-tree-sitter-lang/go/.../Boilerplate.ml *)
+
+(**
+   Boilerplate to be used as a template when mapping the go CST
+   to another type of tree.
+*)
+
+(* Disable warnings against unused variables *)
+[@@@warning "-26-27"]
+
+(* Disable warning against unused 'rec' *)
+[@@@warning "-39"]
+
+[@@@warning "-32"]
+
+type env = H.env
+
+let token (env : env) (_tok : Tree_sitter_run.Token.t) = H.token
+
+let blank (env : env) () = ()
+
+let todo (env : env) _ =
+   failwith "not implemented"
+
+let int_literal (env : env) (tok : CST.int_literal) =
+  token env tok (* int_literal *)
+
+let identifier (env : env) (tok : CST.identifier) =
+  token env tok (* identifier *)
+
+let raw_string_literal (env : env) (tok : CST.raw_string_literal) =
+  token env tok (* raw_string_literal *)
+
+let rune_literal (env : env) (tok : CST.rune_literal) =
+  token env tok (* rune_literal *)
+
+let float_literal (env : env) (tok : CST.float_literal) =
+  token env tok (* float_literal *)
+
+let imaginary_literal (env : env) (tok : CST.imaginary_literal) =
+  token env tok (* imaginary_literal *)
+
+let escape_sequence (env : env) (tok : CST.escape_sequence) =
+  token env tok (* escape_sequence *)
+
+let qualified_type (env : env) ((v1, v2, v3) : CST.qualified_type) =
+  let v1 = token env v1 (* identifier *) in
+  let v2 = token env v2 (* "." *) in
+  let v3 = token env v3 (* identifier *) in
+  todo env (v1, v2, v3)
+
+let package_clause (env : env) ((v1, v2) : CST.package_clause) =
+  let v1 = token env v1 (* "package" *) in
+  let v2 = token env v2 (* identifier *) in
+  todo env (v1, v2)
+
+let empty_labeled_statement (env : env) ((v1, v2) : CST.empty_labeled_statement) =
+  let v1 = token env v1 (* identifier *) in
+  let v2 = token env v2 (* ":" *) in
+  todo env (v1, v2)
+
+let interpreted_string_literal (env : env) ((v1, v2, v3) : CST.interpreted_string_literal) =
+  let v1 = token env v1 (* "\"" *) in
```

Comment:
FWIW I'd love to see us, in the future, clearly separating boilerplate vs. contributed code.

E.g.
```ocaml
match Boilerplate.interpreted_string_literal env v with
  (v1, v2, v3) -> todo env (v1, v2, v3)
```
here

Reply:
The motivation is to reduce, as much as possible, any manual work when rev'ing the grammar later.

Reply:
I don't think we can call Boilerplate from Parse_go_tree_sitter.ml. We copy-paste Boilerplate.ml initially but
in my next diff I will update every final "actions" in Parse_go_tree_sitter.ml. 
When we rev'ing the grammar later, we will regenerate CST.ml and the OCaml compiler will help us pinpoint
all the place in Parse_go_tree_sitter.ml we need to update. I don't think it will be too much work.